src/components/DateInput/types.ts

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { InternalInputColors } from "../internal/InternalInput";
+
+export type DayMode = "none" | "2-digits";
+export type YearMode = "2-digits" | "4-digits";
+
+export interface DateValue {
+  day: string;
+  month: string;
+  year: string;
+}
+
+export interface DeprecatedDateInputProps {
+  name: string;
+  color?: InternalInputColors;
+  label?: string;
+  dayMode?: DayMode;
+  yearMode?: YearMode;
+  helpText?: string;
+  disabled?: boolean;
+  optional?: boolean;
+  validate?: boolean | ((value: any, data: any) => string | string[] | null);
+  validateData?: any;
+  "aria-labelledby"?: string;
+  testId?: string;
+}
+
+export type DateInputError = {
+  day?: string;
+  month?: string;
+  year?: string;
+  field?: string;
+};
```

Comment:
what would be example DateInputError values? trying to understand what field is.

Reply:
This is the error format that is expected return value from validation function